### PR TITLE
Paging control use menu

### PR DESCRIPTION
**Pull request by davewilton**
@rwillsonAxiom I have changed the paging menu to use the menu control. There were a few things I wasn't sure about so I'd appreciate you taking a look and providing some feedback. There are a few " @ruben " in the code.

1. I wanted to center the menu around the anchor rather than left/right align. I did this overriding the constructor of menu and offsetting the anchor by half a menu width.  Is there a better way? Perhaps this could go into the base menu as an option? unless...
2. This is what anchorPosition in the constructor is supposed to be for? I couldn't work out what that was supposed to be doing and I played around with it and it didn't seem to do anything?
3. I didn't do a great job of positioning the left and right buttons. The position is ok, but it uses your pre existing code and assumes the grid is 9x9. Take a look and curse me!
4. I wasn't sure about wall positions, will leave that with you to fix and test
5. Should we make the position of a button optional now as the menus are controlling it?

### CODE REVIEW

**Comment by rwillsonAxiom** [comment]
1. I'm not sure it makes sense in the Menu I made to have a center alignment. I think the way the align booleans work right now is more of a direction flipper. If you want the menu buttons added from right to left you would use `leftAlign = false` etc.
2. The way I made it was thinking that there's a room position that the menu area is anchored to, and the position of the anchor inside that area is the `anchorPosition`.
3. Do you mean 3x3? It looks like they're in a great spot on desktop, and I think it's fine to assume the rows and cols, but I'll make a comment on anything I think can be improved in a minute.
4. All good
5. I think we should. Maybe Menu could have a `createButton` method instead though

File: src/UIControls/MenuPaging.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  private btnPL!: Button;
+  private btnPR!: Button;
+
+  constructor(public width: number, public height: number, public anchor: Vector<3>, public orientation: Quaternion, public anchorPosition: [number, number], public topAligned: boolean, public leftAligned: boolean, public horizontal: boolean) {
+    super(width, height, anchor, orientation, anchorPosition, topAligned, leftAligned, horizontal);
+
+    // recalculate the anchor so that it is centred. @Ruben better way to do this in the base class as an option
+    // @ruben better way to do this?
+    const newX = anchor.data[0] - ((this.cols * this.buttonSize) / 2);
+    anchor.data[0] = newX
```
Comment (rwillsonAxiom) [inline src/UIControls/MenuPaging.ts:29]: I think what you want to do is use an anchorPosition of [0.5, y] to make the anchor coordinates the center of the menu?

File: src/UIControls/MenuPaging.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    this.totalPages = Math.ceil(this.buttons.length / (this.cols * this.rows));
+    this.recomputeButtons = true;
+    this.Draw();
+  }
+  
+
+  private addPagingButtons() {
+    // add the paging buttons to the left and the right
+    const ypr = this.orientation.GetYPR();
+    const newPos = this.getNthButtonPosition(3); // @ruben I fudged this using your code for the 5th button. I'm sure you can do this a better way
```
Comment (rwillsonAxiom) [inline src/UIControls/MenuPaging.ts]: I would suggest using the `getButtonPosition` method instead so you can specify `x = -1 + 0.5` or `x = cols + 0.5` and `y = rows / 2`

**Comment by davewilton** [comment]
@rwillsonAxiom will close this out as discussed in meeting. Thanks for your checks anyway

**Pull request closed without merging**